dev-portal/src/pages/Admin/ApiManagement.jsx

```diff
@@ -232,9 +233,23 @@ export const ApiManagement = observer(class ApiManagement extends React.Componen
     })
   }
 
+  isUpdatingApiGatewayApi (api) {
+    return this.state.apisUpdating.includes(`${api.id}_${api.stage}`)
+  }
+
   updateApiGatewayApi (api) {
+    // Simpler than implementing a multiset, and probably also faster.
+    this.setState(({ apisUpdating }) => ({
+      apisUpdating: [...apisUpdating, `${api.id}_${api.stage}`]
+    }))
     apiGatewayClient()
       .then(app => app.post('/admin/catalog/visibility', {}, { apiKey: `${api.id}_${api.stage}`, subscribable: `${api.subscribable}` }, {}))
+      .then(() => this.setState(({ apisUpdating }) => {
+        const index = apisUpdating.indexOf(`${api.id}_${api.stage}`)
+        const newApisUpdating = apisUpdating.slice()
+        newApisUpdating.splice(index, 1)
+        return { apisUpdating: newApisUpdating }
```

Comment:
Maybe?

```suggestion
        return { apisUpdating: apisUpdating.filter(apiStage => (apiStage !== `${api.id}_${api.stage}`))}
```

Reply:
I specifically only want to remove the *first* occurrence, not all. There could potentially be duplicates here (say, someone clicks it again while waiting) and I want to tolerate it and just fire an extra request in case it somehow updated in that small bit of time. It's simpler than debouncing or trying to detect duplicates.

I'm using an array, but I'm using it more like a `Map<string, Set<Request>>`, where the only part of the value I care about is how many requests it has. And just using an array of strings is easier than implementing a map of anything, much less a `Map<string, number>` or `Map<string, Set<request>>`.

Reply:
Got it.